### ISSUE TITLE
Python-1

### ISSUE THREAD

**Issue by ZLM1986** [ZLM1986/blog#20]
### 環境構築
Python ダウンロード、インストール
pip install XXX
pip list
### 启动jupyter环境
`jupyter lab`

### 包的基本功能一览（按你给的列表）
🌐🧵 网络 / HTTP / 异步
anyio：统一的异步并发抽象层（兼容 asyncio / trio），常被 httpx、jupyter 生态使用
async-lru：异步函数的 LRU 缓存
h11：纯 Python 的 HTTP/1.1 协议实现（httpcore/httpx 底层）
httpcore：HTTP 客户端底层核心（连接池、传输层），供 httpx 使用
httpx：现代 HTTP 客户端（同步/异步、HTTP/2 支持等）
idna：国际化域名（IDN）编码处理
requests：经典同步 HTTP 客户端
urllib3：HTTP 连接池与底层网络库（requests 底层）
websocket-client：WebSocket 客户端实现
🔐🛡️ 加密 / 证书 / 安全
argon2-cffi：Argon2 密码哈希（常用于 Jupyter 登录密码）
argon2-cffi-bindings：argon2 的底层绑定实现
certifi：CA 根证书集合（HTTPS 校验用）
cffi：C Foreign Function Interface（Python 调用 C 库的桥梁）
pycparser：C 语法解析器（cffi 依赖）
defusedxml：安全处理 XML（防 XXE 等攻击）
🧹📄 HTML / 文本清洗与解析
beautifulsoup4：HTML/XML 解析与网页抓取常用库
soupsieve：CSS 选择器支持（BeautifulSoup 的选择器引擎）
bleach：清理/过滤 HTML（防 XSS，nbconvert 等会用）
tinycss2：CSS 解析器（常用于 HTML/CSS 处理链）
webencodings：HTML/CSS 相关编码支持（web 解析链依赖）
mistune：Markdown 解析器（Jupyter/nbconvert 常用）
pandocfilters：Pandoc 过滤器工具（文档转换链）
🌍🈯 国际化 / 时间与日期
babel：国际化与本地化（日期/数字格式化、翻译等）
python-dateutil：日期时间解析与运算增强（pandas 依赖）
arrow：更易用的日期时间处理库
tzdata：时区数据库（在某些系统/环境中提供时区数据）
🧩📦 数据结构 / 校验 / Schema / JSON
attrs：更方便地定义类/数据结构（自动生成比较/初始化等）
jsonschema：JSON Schema 校验
jsonschema-specifications：JSON Schema 规范定义集合（jsonschema 依赖）
referencing：JSON Schema 引用解析工具（jsonschema 相关）
rpds-py：高性能持久化数据结构（jsonschema 相关依赖）
fastjsonschema：更快的 JSON Schema 校验实现（部分场景用）
jsonpointer：JSON Pointer（RFC 6901）实现
json5：JSON5 格式解析（更宽松的 JSON）
PyYAML：YAML 解析与生成
🧠📊 科学计算 / 数据分析 / 机器学习 / 可视化
numpy：数值计算基础（多维数组、向量化运算）
pandas：表格数据处理（DataFrame/Series，清洗、聚合、IO）
scipy：科学计算（优化、统计、信号处理、线性代数等）
scikit-learn：机器学习工具箱（分类/回归/聚类、特征工程、评估）
joblib：并行与模型持久化（sklearn 常用）
threadpoolctl：控制底层 BLAS/OpenMP 线程数（sklearn/scipy 常用）
matplotlib：基础绘图库
seaborn：统计可视化（基于 matplotlib，更高层接口）
contourpy：等高线计算（matplotlib 依赖）
cycler：属性循环（matplotlib 配色/线型循环）
fonttools：字体处理（matplotlib 依赖）
kiwisolver：约束求解器（matplotlib 布局相关依赖）
pillow：图像处理（读写/处理图片，matplotlib 常用）
pyparsing：解析器组合工具（matplotlib 等依赖）
packaging：版本号/依赖规范解析工具
🖥️📓 Jupyter / Notebook 生态（运行、服务端、转换）
jupyterlab：JupyterLab 前端界面
jupyterlab_server：JupyterLab 的服务端组件
jupyter_server：Notebook/Lab 的核心服务端
jupyter_client：与内核通信的客户端协议实现
jupyter_core：Jupyter 核心路径/配置等基础设施
jupyter-events：Jupyter 事件系统与日志/事件记录
jupyter-lsp：Jupyter 的语言服务器协议支持（补全/诊断）
jupyter_server_terminals：在 Jupyter 里提供终端功能
ipykernel：Python 内核（让 Notebook 能运行 Python）
ipython：增强交互式 Python（魔法命令、补全等）
traitlets：配置系统/类型化属性（Jupyter/IPython 基础依赖）
comm：Jupyter 前后端通信通道（widgets/内核通信相关）
nbformat：Notebook 文件格式（.ipynb）读写
nbclient：执行 notebook 的客户端（自动运行 notebook）
nbconvert：Notebook 转换（转 HTML/PDF/Markdown 等）
notebook_shim：兼容层（Notebook 与 Jupyter Server 过渡组件）
terminado：基于 tornado 的终端支持（Jupyter 终端用）
Send2Trash：把文件移到回收站（Jupyter 删除文件用）
prometheus_client：导出监控指标（Jupyter Server 可用）
pyzmq：ZeroMQ 绑定（Jupyter 内核通信底层）
tornado：异步网络框架（Jupyter Server 底层）
pywinpty：Windows 伪终端支持（Jupyter 终端在 Windows 上用）
🧑‍💻🧾 交互、补全、调试、终端显示
prompt_toolkit：交互式命令行输入（IPython 依赖）
Pygments：语法高亮
ipython_pygments_lexers：IPython 的额外语法高亮规则
jedi：代码补全/静态分析（IPython/Jupyter 补全）
parso：Python 解析库（jedi 依赖）
debugpy：调试适配器（VS Code / Jupyter 调试）
decorator：装饰器工具（简化装饰器编写）
colorama：Windows 终端颜色支持
wcwidth：计算字符显示宽度（终端对齐用）
stack-data：生成更友好的堆栈信息（IPython 依赖）
executing：定位“正在执行的表达式”（stack-data 相关）
asttokens：把 AST 节点映射回源码 token（stack-data 相关）
pure_eval：更安全/更纯的表达式求值（stack-data 相关）
🧷🧱 模板 / 标记 / URL 与标识规范
Jinja2：模板引擎（Jupyter/网页渲染常用）
MarkupSafe：安全字符串/转义（Jinja2 依赖）
fqdn：完全限定域名处理/校验（Jupyter 相关校验用）
uri-template：URI Template（RFC 6570）实现
rfc3339-validator：RFC3339 日期时间格式校验
rfc3986-validator：RFC3986 URI 校验
rfc3987-syntax：IRI/URI 语法解析与校验
isoduration：ISO 8601 时长格式解析
🧰⚙️ 安装、构建与运行环境辅助
pip：Python 包安装与依赖管理工具
setuptools：打包/安装基础工具（提供 pkg_resources 等）
wheel：wheel（二进制分发包）构建工具
platformdirs：获取各平台标准配置/缓存/数据目录路径
psutil：系统与进程信息（CPU/内存/进程等）
charset-normalizer：字符编码检测（requests 相关）
six：Py2/3 兼容层（历史遗留依赖，仍常见）
typing_extensions：为旧版本 Python 提供新类型注解特性
python-json-logger：输出 JSON 格式日志
🧩📐 语法/解析相关
lark：通用解析器（写语法、构建解析树）
